File: internal/commands/clear.go
package commands

import (
	"fmt"
	"time"
)

func registerClear(r *Registry) {
	r.Register(&Command{Name: "clear", Description: "Clear all history and start a new session", Kind: Handler,
		Handler: func(args []string, ctx Context) (Result, error) {
			if ctx.Agent != nil {
				ctx.Agent.ClearSession()
				ctx.Agent.SetSession(fmt.Sprintf("session-%d", time.Now().UnixMilli()))
			}
			return StatusResult{Message: "(Cleared)"}, nil
		},
	})
}

File: src/ui/commands/builtin/clear.ts
import { newSessionName } from "../../../services/session-manager.js";
import type { CommandHandler } from "../registry.js";

export const clearCommand: CommandHandler = {
  name: "clear",
  description: "Clear all context and start a new session",
  handler: async (_args, ctx): Promise<void> => {
    ctx.sessionManager.clearSession();
    ctx.contextManager.reset();
    await ctx.switchSession(newSessionName(), { statusMessage: "(Cleared)" });
  },
};
